xml/containers-docker-building-images.xml

```diff
@@ -0,0 +1,460 @@
+   </para>
+<screen>&prompt.user;docker pull registry.suse.com/suse/sle15:15.1</screen>
+  </sect2>
+ </sect1>
+ <sect1 xml:id="Customizing-Pre-build-Images">
+  <title>Customizing &slsa; &docker; Images</title>
+
+  <para>
+   The pre-built images do not have any repositories configured and do not
+   include any modules or extensions. They contain a
+   <link xlink:href="https://github.com/SUSE/container-suseconnect">zypper
+   service</link> that contacts either the &sccreg; or a &rmtool; (&rmt;)
+   server, according to the configuration of the &productname; host that runs
+   the &docker; container. The service obtains the list of repositories
+   available for the product used by the &docker; image. You can also directly
+   declare extensions in your <filename>Dockerfile</filename> (for details
+   refer to
+   <xref linkend="sec-docker-sle-images-customizing-the-images-extensions"/>.
+  </para>
+
+  <para>
+   You do not need to add any credentials to the &docker; image, because the
+   machine credentials are automatically injected into the
+   <literal>/run/secrets</literal> directory in the container by the
+   <systemitem class="daemon">docker</systemitem> daemon. The same applies to
+   the <filename>/etc/SUSEConnect</filename> file of the host system, which is
+   automatically injected into the <filename>/run/secrets</filename> directory.
+  </para>
+
+  <note>
+   <title>Credentials and Security</title>
+   <para>
+    The contents of the <filename>/run/secrets</filename> directory are never
+    included in a &docker; image, hence there is no risk of your credentials
+    leaking.
+   </para>
+  </note>
+
+  <note xml:id="build-images-different-codebase">
+   <title>Building Images on Systems Registered with &rmt;</title>
+   <para>
+    When the host system used for building &docker; images is registered with
+    &rmt;, the default behavior allows only building containers of the same
+    code base as the host. For example, if your &docker; host is an
+    &slea;&nbsp;15 system, you can only build &slea;&nbsp;15-based images on
+    that host by default. To build images for a different &slea; version, for
+    example &slea;&nbsp;12 on an &slea;&nbsp;15 host, the host machine
+    credentials for the target release can be injected into the container as
+    outlined below.
+   </para>
+   <para>
+    When the host system is registered with &scc;, this restriction does not
+    apply.
+   </para>
+  </note>
+
+  <note>
+   <title>Building Container Images in On-Demand &slea; Instances in the Public Cloud</title>
+   <para>
+    Building container images on &slea; instances that were launched as
+    so-called "on-demand" or "pay as you go" instances on a Public Cloud (AWS,
+    GCE, or Azure) requires additional steps. To install packages and updates,
+    the "on-demand" public cloud instances are connected to a public
+    cloud-specific update infrastructure based on &rmt; servers operated by
```

Comment:
This sentence is very long and a little clumsy. How about:

«
To install packages and updates, the "on-demand" public cloud instances are connected to a public cloud-specific update infrastructure. This is based on &rmt; servers operated by...
»